### ISSUE TITLE
[ENH]Add datasets for functional regression

### ISSUE THREAD

**Issue by JonathanBechtel** [sktime/sktime#4314]
**Is your feature request related to a problem? Please describe.**
The collection of datasets available in sktime right now broadly fall into three categories:  univariate forecasting, multivariate classification, and forecasting with exogenous data.  However, there is another type of time series problem which does not have an example dataset:  multivariate regression.  This dataset would have the format `(n_instances, n_timepoints)` in `X`, and a corresponding column of continuous numbers for the target in `y`.  Examples of such data would be a collection of time series from sensor readings with a numeric target.

This issue is problematic because there are a number of estimators in sktime that are built for such use cases and are lacking documentation and examples, presumably because no such example datasets exist.  For example see this GitHub issue: https://github.com/sktime/sktime/issues/4264.

**Describe the solution you'd like**
I'd like to add at least 1-2 datasets that are specifically used for the case of functional regression, where a single sample represents an entire time series and the target is a continuous number.  One example would be the tecator dataset:  https://www.openml.org/search?type=data&sort=runs&id=505&status=active.  These would have similar functionality to existing multi class classification datasets, with a `return_X_y` and `split` arguments.

**Additional context**
You need an adequate number of example datasets to provide adequate documentation for the different estimators in sktime, and without said datasets it will be difficult to patch other issues with useful examples for different estimators.

**Comment by fkiraly** [comment]
Yes, great idea!

Indeed the selection of data sets for tasks other than classification is a bit limited at a moment.
For forecasting, one might for example like to have the M4 or M5 data available. I think @TNTran92 started working on that a while ago.

Summarizing discussion from the Friday collab meeting:

* first step it would be really nice to have regression datasets
* seond step, as you suggested, could be thinking about cleaning up the loader module, adding a clean taxonomy for datasets (e.g., univariate, multivariate, which task) and making them queriable. An example where this is done well is the openML project.

**Comment by JonathanBechtel** [comment]
Yes, correct.  I'll probably create a new issue for this broader agenda to touch up the datasets section, once it's specced out a bit more.  There are a number of tweaks that could be made in order to make it more usable.  

In addition to what you mentioned, here are some other items to think about:
 - Some of the datasets return data in an inconsistent format:  some datasets return `(y, X)`, and others return `(X, y)`
 - Some have a `split` argument, but I'm not sure why others wouldn't.  
 - In general the top level documentation is a little difficult to parse.  I found myself having a hard time getting around to the "right" dataset to use for an example, only to find that it didn't exist.  
 - The UCR_UEA dataset isn't actually a dataset at all, but a loader for any number of datasets from their repository, so it's mistitled in my opinion, and should be changed to `fetch_ucr_uea_dataset` or something similar to denote that it's not a pre-installed dataset but something that actively downloads new data.
 
 Regarding taxonomy, I think a simple improvement would be to have something on the main datasets page akin to this document:  https://docs.google.com/spreadsheets/d/1SNbdgWUoot1tSnuKlxbhKvalYti7L1Z73hnUPXAaZdg/edit?usp=sharing
 
 We can quibble about the details of what to include, but that seems like it'd be an improvement over what's currently here:  http://www.sktime.net/en/latest/api_reference/datasets.html, which is a little sparse on the information you'd actually need to use for an example.  I found myself having to manually download a lot of different ones to verify what was actually inside them.

**Comment by fkiraly** [comment]
Interesting!

I've thought about this a bit:

* I've opened a separate issue here: #4332, I think we should track the general problem of making datasets searchable, findable
* I've drafted an experimental prototype in #4333, which turns datasets into objects with a "load" function. The tags make them searchable via `all_estimators`.

Would appreciate your thoughts! (in #4332, since this issue is more specific about regression)

**Comment by achieveordie** [comment]
I'd also like to bring attention to an old bug that is directly related to this issue - #3521. Essentially, current data loaders are (heavily?) tailored towards classification datasets leading to problems/bugs when trying to load regressional data with it.

**Comment by fkiraly** [comment]
Yes, @achieveordie, I suppose we ought to think carefully how to fix all of this together...

**Comment by hazrulakmal** [comment]
I would like to chip in a bit here. It seems that the majority of the datasets in sktime are pre-installed hence come with sktime when users `pip install sktime`. I recently checked locally and found that the data folder size (where all pre-installed datasets are stored) is around 17.5MB. For context, this is more than half of the whole sktime size (28.6MB). In comparison, our friend from [sklearn dataset](https://scikit-learn.org/stable/datasets/toy_dataset.html) has pre-installed datasets that are less than 300KB and more complicated datasets are fetched externally. 

It would be great to know why storing most of the datasets in the package was made. It would be better if we had only a few pre-installed datasets for each time series task (e.g., forecasting, classification, annotation, etc.), and the remaining datasets would be ported to external resources, like `load_macroeconomics` , `load_solar` and `load_UCR_UEA`. so users can download datasets whenever they want to rather than forcing them to have it when installing sktime. In fact,  all classification preinstalled datasets can be fetched using `load_UCR_UEA`.

In my opinion, having these preinstalled datasets limits in two ways
1. Making the entire package heavy is unnecessary. It is unclear how useful it is to have this pre-installed dataset for users, as most likely they will have their own datasets to work with. much of the utility may probably goes to unit tests or quick demo examples which probably do not need the current number of datasets we have right now.
2. limits sktime ability to provide a real-world and rich collection of datasets (scalability problem). Also, most of the more complicated datasets are actually quite big in size. For example, the raw M5 dataset is about 200MB, while the Rossman dataset is 40 MB. Since recently there was a discussion on benchmarking real-world datasets #4543 It would be nice if we sktime put more focus on having an interface that connects to some of the real-world datasets say from the Monash repo, Rosman Stores Sales or Walmart Sales. users can run one-liner code to download datasets to run benchmarking for example.

A nicer solution would be to have.
1. simple(toy) datasets - (1 or 2 datasets for each task and these pre-installed - must be small say less than 1MB)
2. real-world dataset - which is ported to an external source, then preprocessed internally to a sktime-compatible dataset. Additionally, the existing datasets that are not grouped as simple datasets are also included here. This will have a lot of utils function that deals with downloading datasets from the internet.
3. generated dataset - i.e `_make_hierachiccal`, etc

I think the idea that each of the datasets as a class object for tagging, boilerplate etc is great! In addition to this, would also be great to have a function that automatically searches the dataset object & load it. for example sktime.load_datasets(”insert_dataset_name_here”) then it will automatically load the corresponding datasets.
